tests/queries/0_stateless/02011_tuple_vector_functions.sql

```diff
@@ -82,7 +82,10 @@ SELECT LpNorm((1, 2), toDecimal32(2, 4)); -- { serverError 43 }
 SELECT (1, 2) * toDecimal32(3.1, 8);
 
 SELECT cosineDistance((1, 2), (2, 3, 4)); -- { serverError 43 }
-SELECT tuple() + tuple(); -- { serverError 42 }
+
+-- TODO: what's the expected value of () + ()? Currently it returns 0.
+-- SELECT tuple() + tuple(); -- { serverError 42 }
```

Comment:
I'm not sure. Current implementation returns 0. See https://github.com/amosbird/ClickHouse/blob/29cdc81d6aeed4e374624fa7d329949082d6dd27/src/Functions/vectorFunctions.cpp#L122

Reply:
Maybe that code was dead since we didn't allow `tuple_size == 0`? But it's unclear why result it Int for this case when for any other it's a `Tuple`

Reply:
Yeah. But I think it's fine to leave it undecided for now.

Reply:
In case we return `0` summing up three tuples is not possible, but it's possible for any other tuples:

```sql
:) select tuple() + tuple() as a, toTypeName(a), tuple(1) + tuple(2) as b, toTypeName(b);

┌─a─┬─toTypeName(a)─┬─b───┬─toTypeName(b)─┐
│ 0 │ UInt8         │ (3) │ Tuple(UInt16) │
└───┴───────────────┴─────┴───────────────┘

1 row in set. Elapsed: 0.002 sec. 

ip-172-31-42-13.eu-central-1.compute.internal :) select tuple() + tuple() + tuple();

Received exception from server (version 23.9.1):
Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: Illegal types UInt8 and Tuple() of arguments of function plus: In scope SELECT (tuple() + tuple()) + tuple(). (ILLEGAL_TYPE_OF_ARGUMENT)
```